man/systemd.network.xml

```diff
@@ -307,11 +307,19 @@
         <varlistentry>
           <term><varname>IPv6Token=</varname></term>
           <listitem>
-            <para>An IPv6 address with the top 64 bits unset. When set, indicates the
-            64-bit interface part of SLAAC IPv6 addresses for this link. Note that
-            the token is only ever used for SLAAC, and not for DHCPv6 addresses, even
-            in the case DHCP is requested by router advertisement. By default, the
-            token is autogenerated.</para>
+            <para>Specifies an optional address generation mechanism and an optional address prefix. If
```

Comment:
This documentation is somewhat confusing to me; when eui64 mode is used, the supplied address is not a prefix, it's a suffix. Unless someone is already working on it, I'd be happy to send a PR to improve the documentation.

Reply:
Please do.

Reply:
As I read through the code my confusion is growing... as this 'eui64' mode replaces the existing unnamed 'static' mode where the user supplies the token (which may or may not have been generated using the EUI-64 mechanism). This is even evident in the eui64 test case, where a static token is supplied which is almost certainly not a valid EUI-64 Interface Identifer, but it is treated as such in the code.

Unless I'm mistaken, there should be *three* modes:

* static - user supplies the lowest 64 bits of address, generated in any fashion they wish
* eui64 - user does not (and cannot) supply an address but explicitly requests that an EUI-64 IiD be generated
* prefixstable - user supplies a prefix (of varying length) and an RFC7217 IID is generated

I know I'm coming into this discussion quite late and the PR has already been merged, but I'm currently a happy user of the 'static' mode and I wouldn't want anyone to use the 'eui64' label for that mode, or for any diagnostic or error messages to refer to 'eui64' mode when I didn't request it.

Reply:
@yuwata ?

Reply:
I will take a look later.

Reply:
#14805.